test/HttpServer.test.ts

```diff
@@ -0,0 +1,112 @@
+import { stub, createStubInstance, SinonStubbedInstance } from 'sinon';
+import { use, assert, request, expect } from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import chaiHttp from 'chai-http';
+import {
+    RelayServer,
+    HttpServer,
+    ServerDependencies,
+    TxStoreManager,
+    KeyManager,
+    ServerConfigParams
+} from '../src';
+import { ContractInteractor } from '@rsksmart/rif-relay-common';
+
+use(chaiAsPromised);
+use(chaiHttp);
+
+describe('HttpServer', () => {
+    const fakeServerConfig: Partial<ServerConfigParams> = {
+        url: 'http://localhost:8090'
+    };
+    const port = 8095;
+    const gasPrice = 5;
+    const method: keyof RelayServer = 'getMinGasPrice';
+    let httpServer: HttpServer;
+    let fakeRelayServer: RelayServer;
+    let fakeKeyManager: SinonStubbedInstance<KeyManager>;
+    let fakeContractInteractor: ContractInteractor;
+    let fakeStoreManager: TxStoreManager;
+    let mockDependencies: ServerDependencies;
+
+    before(() => {
+        fakeKeyManager = createStubInstance(KeyManager, {
+            getAddress: 'fakeAddress'
+        });
+        mockDependencies = {
+            managerKeyManager: fakeKeyManager,
+            workersKeyManager: fakeKeyManager,
+            contractInteractor: fakeContractInteractor,
+            txStoreManager: fakeStoreManager
+        };
+        fakeRelayServer = new RelayServer(fakeServerConfig, mockDependencies);
+        httpServer = new HttpServer(port, fakeRelayServer);
+    });
+
+    describe('processRootHandler', () => {
+        it('should process method from relay server', async () => {
+            stub(fakeRelayServer, method).returns(gasPrice);
+            const result = await httpServer.processRootHandler(method, []);
+            assert.equal(result, gasPrice);
+        });
+
+        it('should fail if method does not exist', async () => {
+            const fakeMethod = 'fakeMethod';
+            const error = new Error(
+                `Implementation of method ${fakeMethod} not available on backend!`
+            );
+            await assert.isRejected(
+                httpServer.processRootHandler(fakeMethod, []),
+                error.message
+            );
+        });
+    });
+
+    describe('rootHandler', () => {
+        let fakeRequest;
```

Comment:
Here it may be a good idea to define our specific request type (that is in the `HttpServer.ts`). Something like
```typescript
type RelayRequest = {
   id: string,
   method: string,
   params: Array<unknown>
}
```
I guess?

Reply:
updated